### ISSUE TITLE
Non stable order for keys in CSSModuleExports

### ISSUE THREAD

**Issue by ArnaudBarre** [parcel-bundler/lightningcss#291]
You can play with this [example](https://parcel-css.vercel.app/#%7B%22minify%22%3Afalse%2C%22nesting%22%3Afalse%2C%22customMedia%22%3Atrue%2C%22cssModules%22%3Atrue%2C%22analyzeDependencies%22%3Afalse%2C%22targets%22%3A%7B%22chrome%22%3A6225920%7D%2C%22source%22%3A%22.foo%20%7B%5Cn%20%20color%3A%20blue%3B%5Cn%7D%5Cn%5Cn.bar%20%7B%5Cn%20%20color%3A%20red%3B%5Cn%7D%5Cn%22%2C%22unusedSymbols%22%3A%5B%5D%2C%22version%22%3A%22local%22%7D): Add and remove new lines at the end of the playground and see the CSS modules output changed.

I need this to be able to better cache the JS file resulting from the CSS modules exports. This is not a big issue, I can easily resort the keys in JS.

Unrelated: Maybe the `blue` -> `#00f` transformation should only be applied when minify is on

**Comment by devongovett** [comment]
Rust `HashMap` does not have a stable order, which is why this happens. We could probably move to [indexmap](https://docs.rs/indexmap/latest/indexmap/) assuming there is no perf regression and the binary size increase isn't too big...

**Comment by ArnaudBarre** [comment]
Yes of course if there is a cost for everyone I can ensure the stable on my side
